soc/arm/st_stm32/stm32l4/soc.h

```diff
@@ -71,7 +71,7 @@
 #include <stm32l4xx_ll_rng.h>
 #endif
 
-#if defined(CONFIG_COUNTER_RTC_STM32)
+#if defined(CONFIG_COUNTER_RTC_STM32) || defined(CONFIG_RTC_STM32)
```

Comment:
This should likely be enabled on more SoC if driver is generic enough

Reply:
I'm fine to limit on L4 to start with. There is sufficient work to introduce the api and the driver.
Enabling (and validating) on more SoC could be done later on.